### ISSUE TITLE
✔️ 🚀 Feature Request: Make the max number of columns for a section filterable 

### ISSUE THREAD

**Issue by tomslominski** [elementor/elementor#9406]
## Prerequisites

- [x] I have searched for similar features requests in both open and closed tickets and cannot find a duplicate.
- [x] The feature is still missing in the latest stable version of Elementor ( Elementor Pro. )

**What problem is your feature request going to solve? Please describe.**
Currently, the maximum number of columns is set to a hard limit of 10 within /assets/dev/js/editor/elements/views/section.js. For standard sections with columns this makes sense, but is rather limiting for developers, for example with regards to issue #2218.

**Describe the solution you'd like**
I think the MAX_SIZE variable on line [166](https://github.com/elementor/elementor/blob/master/assets/dev/js/editor/elements/views/section.js#L166) of section.js should be filterable to allow developers to add more columns if their plugin requires. Alternatively, the return value of that whole function could be made filterable to make it more flexible.

**Describe alternatives you've considered**
I've looked at a bunch of other JS filters in Elementor but I can't find any way to override this.

**Additional context**
I'm happy to add a pull request with a fix if you're happy to add this to Elementor.

**Comment by black-eye** [comment]
+1

**Comment by maximseshuk** [comment]
+1

**Comment by hellodracon** [comment]
+1

**Comment by hellodracon** [comment]
That would be great ! Any news on this ? At least for us to be able to break the limit before the update is made ? Thank you

**Comment by melisaciputra** [comment]
Hello! Has anyone figured out the workaround to this?

**Comment by ob2a** [comment]
it should be great…

**Comment by cpflaume** [comment]
+1
Would be really good to be able to change this value. 
I need a flex-grid with 12 elements/columns in it but I can only add 10. This is annoying.

**Comment by yarovojigor1988** [comment]
Changed limit in elementor/assets/js/editor.js to DEFAULT_MAX_COLUMNS = 12; (in elementor/assets/js/editor.min.js changed too) But nothing changed - i can only add 10 columns.

**Comment by hellodracon** [comment]
@yarovojigor1988  You can modify the minified version but not only the var DEFAULT_MAX_COLUMNS = 12. Since it's a minified version, you have to change the value of the variable directly into the code, not the setting of the variable.

Meaning you have to find these lines and change the value by what you want.


isCollectionFilled(){return 10<=this.collection.length} becomes isCollectionFilled(){return 12<=this.collection.length}
and
isEnabled(){return i.model.collection.length<y.DEFAULT_MAX_COLUMNS} isEnabled(){return i.model.collection.length<12}

**Comment by yarovojigor1988** [comment]
Than

> 
> 
> @yarovojigor1988 You can modify the minified version but not only the var DEFAULT_MAX_COLUMNS = 12. Since it's a minified version, you have to change the value of the variable directly into the code, not the setting of the variable.
> 
> Meaning you have to find these lines and change the value by what you want.
> 
> isCollectionFilled(){return 10<=this.collection.length} becomes isCollectionFilled(){return 12<=this.collection.length}
> and
> isEnabled(){return i.model.collection.length<y.DEFAULT_MAX_COLUMNS} isEnabled(){return i.model.collection.length<12}

thanks, now its working

**Comment by hellodracon** [comment]
@yarovojigor1988  remember that if you update elementor, it will update this page too. You can override it (but it's kind of risky) or make the modification after each update (which is safer but time consuming).

**Comment by black-eye** [comment]
> @yarovojigor1988 You can modify the minified version but not only the var DEFAULT_MAX_COLUMNS = 12. Since it's a minified version, you have to change the value of the variable directly into the code, not the setting of the variable.
> 
> Meaning you have to find these lines and change the value by what you want.
> 
> isCollectionFilled(){return 10<=this.collection.length} becomes isCollectionFilled(){return 12<=this.collection.length}
> and
> isEnabled(){return i.model.collection.length<y.DEFAULT_MAX_COLUMNS} isEnabled(){return i.model.collection.length<12}

@hellodracon, I've tried your method, it resolved the the column limit problem, but I'm geting some JS errors when adding or deleting the columns above the original 10 limit.

Example: 
1. I've changed the limit to 20 using the method above.
2. I start to duplicate the columns to reach the limit.
3. Up to 10 columns, everything is fine.
4. Starting with column no. 11, the errors begin to appear.
5. I start to delete the columns.
6. Up to no. 10, new errors keep appearing.
7. Below 10 columns everything is ok againg.

It seems the error is thrown when the content tries to refresh itself. When there's no error, all the columns are redrawing themselves after each deletion. When there's the error, nothing is redrawn (only the deleted column is gone).

Here is the error:
```
common.js?ver=2.9.14:8484 TypeError: Cannot read property '0' of undefined
    at Object.getPresets (editor.js?ver=2.9.14:25536)
    at Object.getPresetByStructure (editor.js?ver=2.9.14:25543)
    at i.adjustColumns (editor.js?ver=2.9.14:5977)
    at editor-document.js?ver=2.9.14:13455
    at Array.forEach (<anonymous>)
    at SetStructure.apply (editor-document.js?ver=2.9.14:13452)
    at SetStructure.run (editor-document.js?ver=2.9.14:6289)
    at Data.runCallback (common.js?ver=2.9.14:10383)
    at Data.runCallbacks (common.js?ver=2.9.14:7119)
    at Data.run (common.js?ver=2.9.14:7080)
```
Any ideas?
Thanks

**Comment by hellodracon** [comment]
@black-eye my solution is just a quick modification to be able to have more than 10 columns but it's not meant to be fully working. I guess you'll have to read a bit of code to find out where is the limitation you're facing.

But maybe, the best solution would be to motivate the dev team to add an input in the setting to set the max column limit and make sure the whole core works with this setting.

**Comment by black-eye** [comment]
@hellodracon Well I've spent several hours trying to read the code, but I had no idea what was wrong, since I'm not super good in JS, especially when things like Webpack are involved. So I tried to ask just in case... ;-)

But thanks for your reply anyway, I appreciate it. It's just your good will helping the others to solve this problem, anyway. Unfortunatelly,  those who should be answering and solving the problems like these just remain in silence. :-(

**Comment by aureliePrd** [comment]
+1 Any news? Please!

**Comment by carlofabyss** [comment]
Also +1, ran into this problem since a client of mine is using a plugin that makes sections into column swiper and the 10 column limit would be nice to be able to change as a filter or anything.  Currently working on a pretty hacky js workaround, not entirely sure it's going to work out, not a big fan of directly editing plugins since they'll be updated at some point independent of me working on it.

**Comment by sgrzegorz** [comment]
@carlofabyss What is this workaroud? I did the modifications in js folder as @hellodracon did, but I still can't add more than 10 columns.

**Comment by nicholaszein** [comment]
Hello everybody!

We have great news! 🙌 

### 📢 We're happy to announce that you can use the <ins>stable</ins> version of the Flexbox Container released with Elementor **v3.8** to add as many child containers as you want! 🥳 

There is no more need to be limited by sections anymore! 💪 

✔️ I invite everyone to check it out and update your plugin to the new version!

Cheers 🥂
